src/lib/connectors/http.js

```diff
@@ -192,7 +192,13 @@ HttpConnector.prototype.request = function(params, cb) {
 
     incoming.setEncoding('utf8');
     incoming.on('data', function(d) {
-      response += d;
+      try {  
+        response += d;
+      } catch (error) {
+        // When stringified response is too big, we will catch this moment
+        // and destroy it to make sure server is NOT crashed
+        this.destroy(error);
```

Comment:
I would override the response at this point 
so we do not keep 500MB in memory which is then dump into the trace log in cleanUp function (when trace log enabled) 
```suggestion
        response = error.message
        this.destroy(error);
```